src/arch/x86_64/syscall_entry.cpp

```diff
@@ -14,63 +14,15 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <arch/x86/cpu.hpp>
 #include <os.hpp>
-#include <likely>
 #include <kprint>
-#include <errno.h>
-
-#define ARCH_SET_GS 0x1001
-#define ARCH_SET_FS 0x1002
-#define ARCH_GET_FS 0x1003
-#define ARCH_GET_GS 0x1004
-
-#ifdef __x86_64__
-static long sys_prctl(int code, uintptr_t ptr)
-{
-  switch(code){
-  case ARCH_SET_GS:
-    //kprintf("<arch_prctl> set_gs to %#lx\n", ptr);
-    if (UNLIKELY(!ptr)) return -EINVAL;
-    x86::CPU::set_gs((void*)ptr);
-    break;
-  case ARCH_SET_FS:
-    //kprintf("<arch_prctl> set_fs to %#lx\n", ptr);
-    if (UNLIKELY(!ptr)) return -EINVAL;
-    x86::CPU::set_fs((void*)ptr);
-    break;
-  case ARCH_GET_GS:
-    os::panic("<arch_prctl> GET_GS called!\n");
-  case ARCH_GET_FS:
-    os::panic("<arch_prctl> GET_FS called!\n");
-  }
-  return -EINVAL;
-}
-#endif
 
 extern "C"
 uintptr_t syscall_entry(uint64_t n, uint64_t a1, uint64_t a2, uint64_t a3,
                    uint64_t a4, uint64_t a5)
 {
-  switch(n) {
-  case 158: // arch_prctl
-    sys_prctl(a1, a2);
-    break;
-  default:
-    kprintf("<syscall entry> no %lu (a1=%#lx a2=%#lx a3=%#lx a4=%#lx a5=%#lx) \n",
+  kprintf("<syscall entry> no %lu (a1=%#lx a2=%#lx a3=%#lx a4=%#lx a5=%#lx) \n",
```

Comment:
Should we panic here? I don't think we ever want to use the syscall instruction so if some library does (like if libc decides to introduce it in more places) this will loop pretty mysterious. Not a dealbreaker though - this is what the kernel is doing today, so it's fine. But if you agree maybe use the opportunity to fail early.